### PR TITLE
moved header footer and side nav to npm

### PR DESCRIPTION
**Pull request by srikanthgudipudi**

### CODE REVIEW

File: projects/shikshalokam/src/lib/core-module/components/headers/headers.component.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<mat-toolbar class="csidenav">
+    <div style="padding-top: 10px;">
+        <img class="b-color" style="width: 15%;height: 10%;" alt="Angular Logo" src="assets/images/shikshalokam.png" />
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/components/headers/headers.component.html]: use class

File: projects/shikshalokam/src/lib/core-module/components/headers/headers.component.scss [hunk 2]
```diff
@@ -0,0 +1,26 @@
+mat-sidenav {
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/components/headers/headers.component.scss]: use of this class??

File: projects/shikshalokam/src/lib/core-module/components/headers/headers.component.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { Component, OnInit,Input, Output, EventEmitter } from '@angular/core';
+
+@Component({
+  selector: 'lib-headers',
+  templateUrl: './headers.component.html',
+  styleUrls: ['./headers.component.scss']
+})
+export class HeadersComponent implements OnInit {
+  username: any;
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/components/headers/headers.component.ts:9]: Where do we initialize this variable??

File: projects/shikshalokam/src/lib/core-module/components/side-nav/side-nav.component.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<mat-sidenav-container fullscreen>
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/components/side-nav/side-nav.component.html]: Why is this container here???

File: projects/shikshalokam/src/lib/core-module/components/side-listing/side-listing.component.html [hunk 5]
```diff
@@ -0,0 +1 @@
+<p>side-listing works!</p>
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/components/side-listing/side-listing.component.html]: Why is there no code??

File: projects/shikshalokam/src/lib/core-module/constants/menu.interface.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export interface MenuData {
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/constants/menu.interface.ts]: Where are you using this?? This should be in the interface folder.

File: projects/shikshalokam/src/lib/core-module/components/headers/headers.component.html (hunk 1 above)
Comment (srikanthgudipudi) [reply projects/shikshalokam/src/lib/core-module/components/headers/headers.component.html]: changed to class

File: projects/shikshalokam/src/lib/core-module/components/headers/headers.component.scss (hunk 2 above)
Comment (srikanthgudipudi) [reply projects/shikshalokam/src/lib/core-module/components/headers/headers.component.scss]: we are over riding this css with our custom width

File: projects/shikshalokam/src/lib/core-module/components/headers/headers.component.ts (hunk 3 above)
Comment (srikanthgudipudi) [reply projects/shikshalokam/src/lib/core-module/components/headers/headers.component.ts:9]: In the same component

File: projects/shikshalokam/src/lib/core-module/components/side-nav/side-nav.component.html (hunk 4 above)
Comment (srikanthgudipudi) [reply projects/shikshalokam/src/lib/core-module/components/side-nav/side-nav.component.html]: if i place inside the <mat-sidenav-container> in portal and <mat-sidenav> in library it is not working. that is why i kept in library only

File: projects/shikshalokam/src/lib/core-module/components/side-listing/side-listing.component.html (hunk 5 above)
Comment (srikanthgudipudi) [reply projects/shikshalokam/src/lib/core-module/components/side-listing/side-listing.component.html]: if we are passing data from the particular portal itself right ? then again why we need to maintain the listing here

File: projects/shikshalokam/src/lib/core-module/constants/menu.interface.ts (hunk 6 above)
Comment (srikanthgudipudi) [reply projects/shikshalokam/src/lib/core-module/constants/menu.interface.ts]: changed

File: projects/shikshalokam/src/lib/core-module/components/footer/footer.component.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  selector: 'lib-footer',
+  templateUrl: './footer.component.html',
+  styleUrls: ['./footer.component.scss']
+})
+export class FooterComponent implements OnInit {
+  footerLink
+  constructor() {
+    this.footerLink =[
+      {
+        name: "Copyright @2020 Shikshalokam"
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/components/footer/footer.component.ts]: get the year from  Javascript date

File: projects/shikshalokam/src/lib/core-module/components/footer/footer.component.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+      },
+      {
+        name: "Terms of Service"
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/components/footer/footer.component.ts]: Keep one more field as URL  and take the URLs from shikshalokam website. On clicking open the link in new tab

File: projects/shikshalokam/src/lib/core-module/components/headers/headers.component.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+      this.username = this.logindetails.username;
+    }
+  }
+
+  notifications() {
+
+  }
+
+  logout() {
+    this.logoutClick.emit('logout');
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/components/headers/headers.component.ts:28]: Logout we can handle here itself.. You can pass the logout URL to here

File: projects/shikshalokam/src/lib/core-module/components/page-layout/page-layout.component.ts [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+export class PageLayoutComponent implements OnInit {
+  type: any;
+  @Input() sideMenuData: MenuData[];
+  @Output() sideMenuClick = new EventEmitter();
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+  clicked(data) {
+    console.log('PageLayoutComponent - clicked');
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/components/page-layout/page-layout.component.ts]: remove unwanted logs

File: projects/shikshalokam/src/lib/core-module/components/footer/footer.component.ts (hunk 7 above)
Comment (srikanthgudipudi) [reply projects/shikshalokam/src/lib/core-module/components/footer/footer.component.ts]: Changed

File: projects/shikshalokam/src/lib/core-module/components/footer/footer.component.ts (hunk 8 above)
Comment (srikanthgudipudi) [reply projects/shikshalokam/src/lib/core-module/components/footer/footer.component.ts]: Added

**Pull request merged**